x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/TransformIT.java

```diff
@@ -225,6 +233,121 @@ public void testBasicContinuousTransformStats() throws Exception {
         deleteTransform(transformId);
     }
 
+    public void testTransformLifecycleInALoop() throws Exception {
+        String transformId = "lifecycle-in-a-loop";
+        String indexName = transformId + "-src";
+        createReviewsIndex(indexName, 100, NUM_USERS, TransformIT::getUserIdForRow, TransformIT::getDateStringForRow);
+
+        String destIndex = transformId + "-dest";
+        String config = createConfig(transformId, indexName, destIndex);
+        for (int i = 0; i < 100; ++i) {
+            long sleepAfterStartMillis = randomLongBetween(0, 5_000);
+            boolean force = randomBoolean();
+            try {
+                // Create the continuous transform
+                putTransform(transformId, config, RequestOptions.DEFAULT);
+                assertThat(getTransformTasks(), is(empty()));
+                assertThatTransformTaskDoesNotExist(transformId);
+
+                startTransform(transformId, RequestOptions.DEFAULT);
+                // There is 1 transform task after start
+                assertThat(getTransformTasks(), hasSize(1));
+                assertThatTransformTaskExists(transformId);
+
+                Thread.sleep(sleepAfterStartMillis);
+                // There should still be 1 transform task as the transform is continuous
+                assertThat(getTransformTasks(), hasSize(1));
+                assertThatTransformTaskExists(transformId);
+
+                // Stop the transform with force set randomly
+                stopTransform(transformId, true, null, false, force);
+                // After the transform is stopped, there should be no transform task left
+                assertThat(getTransformTasks(), is(empty()));
+                assertThatTransformTaskDoesNotExist(transformId);
+
+                // Delete the transform
+                deleteTransform(transformId);
+            } catch (AssertionError | Exception e) {
+                throw new AssertionError(
+                    format("Failure at iteration %d (sleepAfterStartMillis=%s,force=%s): %s", sleepAfterStartMillis, force, e.getMessage()),
```

Comment:
There are 4 wildcards but only 3 parameters, I think the `i` is missing?

```suggestion
                    format("Failure at iteration %d (sleepAfterStartMillis=%s,force=%s): %s", i, sleepAfterStartMillis, force, e.getMessage()),
```